src/components/screens/consul/Proposals/NewProposal.js

```diff
@@ -270,7 +318,92 @@ export const NewProposal = ({ navigation, data, query }) => {
   );
 };
 
+const ImageSelector = ({ control, field, item, selectImage }) => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [imageInfoText, setImageInfoText] = useState('');
+
+  const { buttonTitle, infoText } = item;
+  const { name, onChange, value } = field;
+
+  return (
+    <>
+      <Input
+        {...item}
+        control={control}
+        errorMessage={errorMessage}
+        hidden
+        validate
+        name={name}
+        value={value}
+      />
+      <RegularText smallest placeholder>
+        {infoText}
+      </RegularText>
+
+      {value ? (
+        <>
+          <WrapperRow center spaceBetween>
+            <Image source={{ uri: value }} style={styles.image} />
+
+            <TouchableOpacity
+              onPress={() => {
+                onChange('');
+                setErrorMessage('');
+              }}
+            >
+              <Icon.Trash color={colors.error} size={normalize(16)} />
+            </TouchableOpacity>
+          </WrapperRow>
+          <RegularText smallest>{imageInfoText}</RegularText>
+        </>
+      ) : (
+        <Button
+          title={buttonTitle}
+          invert
+          onPress={async () => {
+            const { uri, type } = await selectImage();
+            const { size } = await FileSystem.getInfoAsync(uri);
+            const errorMessage = imageErrorMessageGenerator({
+              size,
+              uri: uri.substr(uri.length - 4)
+            });
+
+            setErrorMessage(texts.consul.startNew[errorMessage]);
+            setImageInfoText(`(${type}/${uri.substr(uri.length - 3)}, ${bytesToSize(size)})`);
```

Comment:
what if the filetype ending is not only 3 chars? like jpeg? this is very unflexible.
doesn't `FileSystem.getInfoAsync(uri)` return some filetype stuff? i would think so and we would not need to build the string on our own.

Reply:
sorry, i can't find the type of the image
type comes only as `image` or `video`.

`selectImage()` :
```
Object {
  "cancelled": false,
  "height": 800,
  "type": "image",
  "uri": "file:///Use......0EE9A6.png",
  "width": 800,
}
```

`FileSystem.getInfoAsync(uri)` :
```
Object {
  "exists": true,
  "isDirectory": false,
  "modificationTime": 1652857258.6079366,
  "size": 239169,
  "uri": "file:///Use......2C660.jpg",
}
```

Reply:
I just take the last 3 letters of the uri and determine the type of the picture. In another way, the picture type does not reach me.